### ISSUE TITLE
Support Python3

### ISSUE THREAD

**Issue by simsong** [getdnsapi/getdns-python-bindings#31]
Python3 is preferred because of its improved Unicode support and future.

**Comment by MelindaShore** [comment]
Python3 is already supported in the develop branch.  I'm planning on getting a new release out by the end of August, but in the meantime if you need Python3 support please check out the develop branch and build with "python3 setup.py {build,install}"

**Comment by simsong** [comment]
That doesn't seem to work:
```
[ec2-user@ec2 getdns-python-bindings]$ git checkout develop
Branch develop set up to track remote branch develop from origin.
Switched to a new branch 'develop'
[ec2-user@ec2 getdns-python-bindings]$ python34 setup.py build
running build
running build_ext
building 'getdns' extension
creating build/temp.linux-x86_64-3.4
gcc -pthread -Wno-unused-result -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/local/include -I/usr/include/python3.4m -c getdns.c -o build/temp.linux-x86_64-3.4/getdns.o -g
getdns.c:38:20: fatal error: Python.h: No such file or directory
 #include <Python.h>
                    ^
compilation terminated.
error: command 'gcc' failed with exit status 1
[ec2-user@ec2 getdns-python-bindings]$ 
```

The `#include <Python.h>` should only be used in a Python2 program, not a python3.

**Comment by MelindaShore** [comment]
Hi, Simson - that's actually not the case.  If you look at the developer documentation for 3.4.3, it says

> Note: Since Python may define some pre-processor definitions which affect the standard headers on some systems, you must include Python.h before any standard headers are included.

in section 1.1 (see https://docs.python.org/3/extending/extending.html).  Please make sure that you've got python3-dev (or the appropriate development libraries and header files for your OS) installed and give it another go.  

In the meantime I'm reopening this until we've confirmed that it compiles and runs for you.

**Comment by simsong** [comment]
Hm. Now I see:

```
gcc -pthread -Wno-unused-result -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/local/include -I/usr/include/python3.4m -c getdns.c -o build/temp.linux-x86_64-3.4/getdns.o -g
In file included from getdns.c:47:0:
pygetdns.h:94:5: error: unknown type name ‘getdns_transport_list_t’
     getdns_transport_list_t *dns_transport_list;
     ^
error: command 'gcc' failed with exit status 1
[ec2-user@ec2 getdns-python-bindings]$ 
```

I'm running getdns-0.2.0; will upgrade to 0.3.1 and try again.

Simson

**Comment by MelindaShore** [comment]
Released in v0.4.0
